Bottom: 46615b7edd3b4a8561c0e4419d89bdeb88db2e30
Top:    25d50cd3051d3f2dd1864415bb489d3f72afdb61
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-18 16:36:07 -0700

rsockets: Define options specific to rsockets

Allow a user to control some of the RDMA related attributes
of an rsocket through setsockopt/getsockopt.  A user specifies
that the rsocket should be modified through SOL_RDMA level.

This patch provides the initial framework.  Subsequent patches
will add the configurable parameters.

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/include/rdma/rsocket.h b/include/rdma/rsocket.h
index 5310d33..87ee943 100644
--- a/include/rdma/rsocket.h
+++ b/include/rdma/rsocket.h
@@ -72,6 +72,8 @@ int rselect(int nfds, fd_set *readfds, fd_set *writefds,
 int rgetpeername(int socket, struct sockaddr *addr, socklen_t *addrlen);
 int rgetsockname(int socket, struct sockaddr *addr, socklen_t *addrlen);
 
+#define SOL_RDMA 0x10000
+
 int rsetsockopt(int socket, int level, int optname,
 		const void *optval, socklen_t optlen);
 int rgetsockopt(int socket, int level, int optname,
diff --git a/src/rsocket.c b/src/rsocket.c
index 4e0671f..15e28de 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1690,6 +1690,8 @@ int rsetsockopt(int socket, int level, int optname,
 			break;
 		}
 		break;
+	case SOL_RDMA:
+		break;
 	default:
 		break;
 	}
@@ -1755,6 +1757,9 @@ int rgetsockopt(int socket, int level, int optname,
 			break;
 		}
 		break;
+	case SOL_RDMA:
+		ret = ENOTSUP;
+		break;
 	default:
 		ret = ENOTSUP;
 		break;
